### ISSUE TITLE
Release MathSE ORM and Reflection Dataset on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [zheny2751-dotcom/MathSE#1]
Hi @ZhenYang21 🤗

Niels here from the open-source team at Hugging Face. I discovered your work through Hugging Face's daily papers as yours got featured: https://huggingface.co/papers/2511.06805.
The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

It's fantastic to see that the main MathSE models (CogVLM2, Qwen2-VL-7B, InternVL2.5-8B backbones) are already available on the 🤗 Hub at `https://huggingface.co/ZhenYang21/MathSE`! We noticed that they are not yet linked to your paper page on our platform, and we'd be happy to help you with that to further improve their discoverability.

Additionally, I saw on your project page that the "Outcome Reward Model checkpoints" and the "Reflection dataset + prompts" are listed as "Coming Soon". It'd be great to make these additional checkpoints and dataset available on the 🤗 hub too, to improve their discoverability/visibility.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models (Outcome Reward Model)

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

You can also build a demo for your model on [Spaces](https://huggingface.co/spaces), we can provide you a ZeroGPU [grant](https://huggingface.co/docs/hub/en/spaces-gpus#community-gpu-grants), which gives you A100 GPUs for free.

## Uploading dataset (MathSE Reflection Dataset)

Would be awesome to make the MathSE Reflection Dataset available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

After uploaded, we can also link the datasets to the paper page (read [here](https://huggingface.co/docs/hub/en/model-cards#linking-a-paper)) so people can discover your work.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗
